src/cheri_riscv_types.sail

```diff
@@ -24,27 +24,46 @@ function ext_accum_ptw_result(ptw : ext_ptw, pte : pte_check) -> ext_ptw =
   }
 
 /* Address translation errors */
-enum ext_ptw_error = {AT_LOAD_CAP_ERR, AT_STORE_CAP_ERR}
+enum ext_ptw_error = {AT_CAP_ERR}
 
 /* CHERI exception extensions */
 
-enum ext_exc_type = {EXC_CHERI, EXC_CHERI_VMEM_LOAD_CAP, EXC_CHERI_VMEM_STORE_CAP}
+enum ext_exc_type = {
+  EXC_LOAD_CAP_PAGE_FAULT,
+  EXC_SAMO_CAP_PAGE_FAULT,
+  EXC_CHERI
+}
 
 /* CHERI translation of PTW errors into exception annotations */
 function ext_translate_exception(e : ext_ptw_error) -> ext_exc_type =
   match e {
-    AT_LOAD_CAP_ERR  => EXC_CHERI_VMEM_LOAD_CAP,
-    AT_STORE_CAP_ERR => EXC_CHERI_VMEM_STORE_CAP
-}
+    AT_LOAD_CAP_ERR  => EXC_LOAD_CAP_PAGE_FAULT,
+    AT_STORE_CAP_ERR => EXC_SAMO_CAP_PAGE_FAULT
+  }
 
 /* CHERI conversion of extension exceptions to bits */
 val ext_exc_type_to_bits : ext_exc_type -> exc_code
-function ext_exc_type_to_bits(e) = 0x1C
+function ext_exc_type_to_bits(e) =
+  match (e) {
+    EXC_LOAD_CAP_PAGE_FAULT => 0x1a,
+    EXC_SAMO_CAP_PAGE_FAULT => 0x1b,
+    EXC_CHERI               => 0x1c
+  }
 
 /* CHERI conversion of extension exceptions to integers */
 val num_of_ext_exc_type : ext_exc_type -> {'n, (0 <= 'n < xlen). int('n)}
-function num_of_ext_exc_type(e) = 28
+function num_of_ext_exc_type(e) =
+  match (e) {
+    EXC_LOAD_CAP_PAGE_FAULT => 26,
+    EXC_SAMO_CAP_PAGE_FAULT => 27,
+    EXC_CHERI               => 28
+  }
```

Comment:
I can't help but feel that one of `num_of_ext_exc_type` and `ext_exc_type_to_bits` should be redundant.  (This presumably requires a change in `riscv-sail` as well).

Reply:
I had the same thought, it's absurd. But yes, sail-riscv needs to change for that.